src/zeta/aca_zeta_programming.cpp

```diff
@@ -26,50 +30,99 @@ ACA_Zeta_Programming &ACA_Zeta_Programming::get_instance()
   return instance;
 }
 
-int ACA_Zeta_Programming::update_zeta_group_entry(zeta_config *zeta_config_in)
+int ACA_Zeta_Programming::update_zeta_config(const alcor::schema::VpcState current_VpcState)
 {
-  unsigned long not_care_culminative_time;
+  zeta_config stZetaCfg;
   int overall_rc = EXIT_SUCCESS;
 
-  list<struct destination>::iterator it;
+  string vpc_id = current_VpcState.configuration().id();
+
+  AuxGateway current_ZetaState = current_VpcState.configuration().auxiliary_gateway();
+  if (current_ZetaState.auxgateway_type() == 0) {
+    ACA_LOG_INFO("%s", "AuxGateway_type is NONE!\n");
+  }
+
+  stZetaCfg.group_id = current_ZetaState.id();
+  for (auto destination : current_ZetaState.destinations()) {
+    stZetaCfg.zeta_buckets.push_back(destination.ip_address());
+  }
+  uint32_t oam_server_port = current_ZetaState.zeta_info().port_inband_operation();
+
+  //I am not sure if the logic of this part is correct?
```

Comment:
vpc configuration is to provide additional information on port operations (CREATE/DELETE/etc). Therefore you want process ZGC information during the processing of port. Let me finish adding the support for port delete and merge my change first, and then you will make change on top of it. 

With that said, you won't need to do any processing on VPC configuration.